### ISSUE TITLE
Avoid op_ref in macros

### ISSUE THREAD

**Issue by zayenz** [rust-lang-nursery/rust-clippy#2818]
The `op_ref` lint may give false positives when checking macros. Given the macro
```
macro_rules! find {
    ($app:expr, $name:expr) => {
        $app.args.iter().find(|a| &a.name == $name)
    }
}
```
and the call
```
let arg = find!(self.app, &name).expect(INTERNAL_ERROR_MSG);
```
clippy gives the suggestion
```
warning: needlessly taken reference of both operands
    --> src/macros.rs:1023:35
     |
1023 |         $app.args.iter().find(|a| &a.name == $name)
     |                                   ^^^^^^^
     | 
    ::: src/app/parser.rs:467:35
     |
467  |                         let arg = find!(self.app, &name).expect(INTERNAL_ERROR_MSG);
     |                                   ---------------------- in this macro invocation
     |
     = note: #[warn(op_ref)] on by default
     = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.205/index.html#op_ref
```

The macro is used in many different contexts, and in many of theses cases the argument is already a reference. Thus, the lint should probably be turned off for macros.

Nightly version: nightly-x86_64-apple-darwin rustc 1.28.0-nightly (990d8aa74 2018-05-25)
Clippy version: 0.0.205
Code that triggers the error: https://github.com/kbknapp/clap-rs/tree/v3-master at commit 3b633e68b16645f1f79e1856bab7adcc522b952e

Note that the above code also results in an ICE (#2689), but the lint is produced before that.

**Comment by zayenz** [comment]
If this is of interest, then the branch https://github.com/zayenz/rust-clippy/tree/no-op-ref-in-macro implements the change.

Interestingly, this change avoids the issue in #2689.

**Comment by oli-obk** [comment]
Always feel free to just open PRs! That makes reviewing much easier for us and usually we just accept this kind of stuff easily

**Comment by oli-obk** [comment]
fixed in #2819
